### PR TITLE
Adding a failure note for decompression failure

### PR DESCRIPTION
**Pull request by martinthomson**
Proposal to replace #696.  I think that this is a little clearer this way.

Also, it's strange that this was not added earlier, but then sometimes the most obvious things get missed.

cc @RobbySimpson

### CODE REVIEW

File: draft-ietf-httpbis-http2.xml [hunk 1]
```diff
@@ -716,7 +716,12 @@ HTTP2-Settings    = token68
         </t>
         <t>
           Header compression is stateful.  One compression context and one decompression context is
-          used for the entire connection.  Each header block is processed as a discrete unit.
+          used for the entire connection.  Failure to decompress a header
+          block MUST be treated as a <xref target="ConnectionErrorHandler">connection error</xref>
+          of type <x:ref>COMRPESSION_ERROR</x:ref>.
```
Comment (ijl) [inline draft-ietf-httpbis-http2.xml]: @martinthomson, I see there's a typo on `COMRPESSION_ERROR`.

**Comment by RobbySimpson** [comment]
So I'm not seeing how the new sentence is any different than this one two paragraphs below:
A receiver MUST terminate the connection with a connection error of type COMPRESSION_ERROR if it does not decompress a header block.

My chief concern (merely editorial) is that HPACK consistently uses the phrase "decoding error" whereas here we are using the phrase "decompress."  I think it good to make clear that any and all decoding errors MUST result in connection errors of type COMPRESSION_ERROR.

Picky, I know.  ;-)

**Comment by hruellan** [comment]
I think that Martin's proposal tries to specifically address your concern.

The sentence two paragraphs below address the case of an endpoint deciding for whatever reason not to decompress a header block: this also has to result in a COMPRESSION_ERROR.

So I think that both are useful.

**Comment by RobbySimpson** [comment]
Ahhh - thanks @hruellan!  Makes perfect sense and I agree.

**Pull request merged**